### ISSUE TITLE
Prepare for adaptation options calculations

### ISSUE THREAD

**Issue by DenoBeno** [clarity-h2020/csis#154]
- [x] Change the "costs" fields to free text (cost per m^2)
- [x] Assure that the new cost fields are shown in all relevant views
- [x] [Implement new data model (node) for "adaptation strategy"](https://github.com/clarity-h2020/csis/issues/155) . Each strategy is for ONE land use; three variants to be created for each land use 
- [ ] Allow user to define the "adaptation project" by choosing several adaptation strategies, one for each land use category (user explicitly has to choose "no adaptation" or one of the options.
- [ ] **TBD: can the user define more than one adaptation project or just one?**
- [ ] calculate the effects of adaptation options
- [ ] calculate the costs of adaptation options 
- [ ] calculate the cost of hazard impact with/without adaptation options.
- [ ] present the results to the user.

**Comment by DenoBeno** [comment]
After yesterdays discussion, I have a better Idea how to proceed with this. First, we need **change in the "cost" field of the adaptation options**. There, the three cost fields need to be replaced with numeric fields (price pro m^2). This is easy, but we also need to change a presentation in all views to accommodate these three new fields and then drop the old ones. 

@stefanon, @mattia-leone  : I hope that I can change this later today.

Once the fields are available, PLINIUS will start filling in the data. All values will be in €, valid for Italy (we need to inform the user about this and also to think about transforming this to Austria or Sweden later.)

- Considering the issue with costs of per-meter adaptation options, we can ignore these. All adaptation options will have per m^2 cost, how is up to PLINIUS.

Next is to define a **new node or taxonomy type that will be used to define "adaptation strategy"**. Something like this:
1. one **land use category** (e.g. residential middle density houses)
2. one or or more **adaptation options**
3. **percentage** of the land use category to apply this to (one or one for each adaptation option?)
4. "strategy" (need another name for this) of choosing where to apply this - e.g. random or "cells with  high heat hazard" or "cells with high population exposure" or "cells with high heat mortality"
5. some text & illustrations
6. Other?

Each adaptation strategy defines one or more adaptation options applied to one land use category. E.g. "apply cool roofs and green walls on 50% of the residential middle density houses". _I am not 100% sure what exactly is needed in this node, so I'll open a second ticket for this._

- As soon as we have this covered, PLINIUS will add some strategies. 

Third part is to add "Adaptation Options: Appraisal" step where user can **choose _one_ adaptation option strategy for each land use**. They don't have to choose adaptation strategy for each land use category - those that aren't set are implicitly set to "no adaptation option".

I'm thinking of some table where users can choose this. The simplest solution would be to replicate what we already do for adaptation options on https://csis.myclimateservice.eu/study/25/step/3686/view/table. The problem here is that nothing prevents the user to choose several conflicting options. Better solution would be a table listing all land uses in the first column and allowing the user to choose the adaptation strategy in the second, plus having some information about this strategy shown in the third column. @patrickkaleta , @fgeyer16, do you have some bright ideas on how to do this? (I'll probably open new ticket to decide on how exactly to do this.)

----------
@humerh : this makes it clear what EMIKAT can get from CSIS. A list of "strategies" to apply. Each strategy is linked to its description, and then to individual adaptation options. There is no explicit geometry, just the geometry of the project. The difficult part is that EMIKAT would have to choose where to apply the the "strategy". E.g., if the strategy is to apply something to 50% of the houses, preferably in the "cells with high mortality", then EMIKAT would needs to find how many houses are in the project, find the cell with highest mortality, apply this to all houses in this cell, find the next highest mortality cell, apply there and so on - until 50% mark has been reached. 

This sounds too complex to me, we need to re-design this somehow. @all, any ideas?
---------
As for the cost calculations, I don't know what we have at a table today, if anything. @mattia-leone , @stefanon  How are we (going to) calculate the cost of the crisis? @humerh , @patrickkaleta: is anything already designed or implemented in this direction?

---------
Finally, we will need something similar to what is already shown in risk/impact and ideally a possibility to compare a situation with/without adaptation options. Which reminds me that we'll have to calculate the cumulative costs over N years, not just costs of a single event.

Any comments/sugestions?

**Comment by mattia-leone** [comment]
Dear all, I am a bit lost. Everything was clear at the time of my last meeting with Denis about how to calculate the effect of adaptation, using "strategies" instead of single measures, and working with the land use percentages.
At this point I think that we should have a dedicated meeting to talk about this, let's not stall with this crucial step!

**Comment by mattia-leone** [comment]
about adaptation cost calculations, we have a very complete set of data about this. 
we agreed that first the data model for strategies on the CSIS should be prepared, so that we can input all the data.
Also, we were waiting for someone to modify the cost field for the single adaptation measures (insted of €, €€, €€€) so that we can insert the revised cost data

**Comment by patrickkaleta** [comment]
> about adaptation cost calculations, we have a very complete set of data about this.
> we agreed that first the data model for strategies on the CSIS should be prepared, so that we can input all the data.
> Also, we were waiting for someone to modify the cost field for the single adaptation measures (insted of €, €€, €€€) so that we can insert the revised cost data

I changed the cost fields, so you should be able to now insert specific values for development, retrofitting and maintenance costs. Remaining changes to the data structure will be done after the initial release of the CSIS, because such changes always affect the content synchronization between Development site and production site, which I want to keep at a minimum before the initial release.

Just one question, so that I can add a correct description to the cost fields. Is the maintenance cost calculated on a per-year basis or for the entire estimated lifespan of the adaptation option?

**Comment by DenoBeno** [comment]
I see that Partrick has changed the cost fields in adaptation options, but the old fields (and not the new ones) are still shown everywhere. 

I will open a new issue for this and proceed with fixing the views.

**Comment by humerh** [comment]
The interface to the REST interface, including adaptation options is included in the Wiki page:
https://github.com/clarity-h2020/csis/wiki/Services-endpoints-(used-by-CSIS)

The new views for cost reports will also be documented there.

**Comment by patrickkaleta** [comment]
CSIS is now able to show the results for the Adaptation option calculations in the **Map** and **Table** component. The Adaptation Strategy selected by the user is linked with the **study_variant** keyword `ADAPTATION-01`. 

So, depending on the currently selected **Study scenario**, the Map and Table components will pull the results for either the baseline scenario or the adapted scenario from Emikat.

**Example:** https://csis-dev.myclimateservice.eu/study/25/step/3682/view/table (selecting either "**worst case, 20y, adapted**" or "**worst case, 20y**" will show different results.

**Comment by patrickkaleta** [comment]
Ideally, we can decouple the study_variant from the Study Scenarios, so that it could be possible to show in the Map component the differences of the Study scenario with and without Adaptation options side by side (we already have the split screen with two maps, so we should use it). @therter let's maybe discuss this in a short telco to see how feasible that would be for us in the remaining time.

**Comment by patrickkaleta** [comment]
### Another thing, we need to check the plausibility of the adaptation options calculations. 
I'm afraid currently it's not quite what we are looking for. For example see these two layers for the "Local effect mean radiant temperature":
1) ["No adaptation options" layer](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27BASELINE%27%20and%20TIME_PERIOD=%2720710101-21001231%27%20AND%20EMISSIONS_SCENARIO=%27rcp85%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&layers=clarity%3Aview.2974&styles=T_MRT%0D%0A&format=image%2Fpng&transparent=true&version=1.1.1&width=1536&height=1536&srs=EPSG%3A3857&bbox=1800244.8901724715,6124746.202434603,1858948.527895487,6183449.840157619)
2) ["with adaptation options" layer](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27ADAPTATION-01%27%20and%20TIME_PERIOD=%2720710101-21001231%27%20AND%20EMISSIONS_SCENARIO=%27rcp85%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&layers=clarity%3Aview.2974&styles=T_MRT%0D%0A&format=image%2Fpng&transparent=true&version=1.1.1&width=1536&height=1536&srs=EPSG%3A3857&bbox=1800244.8901724715,6124746.202434603,1858948.527895487,6183449.840157619)

Otherwise, the selected Study scenarios are the same (same time period, RCP...), so the "with adaptation options" layer should show less red squares and not more.

Also, in the [Table ](https://csis-dev.myclimateservice.eu/study/25/step/3682/view/table), the adapted scenario shows higher temperatures:
![baseline-temperatures](https://user-images.githubusercontent.com/42167036/84996456-97775600-b14d-11ea-969f-3ff7648edad8.png)
![adapted-temperatures](https://user-images.githubusercontent.com/42167036/84996463-99d9b000-b14d-11ea-8048-fc5f3a9c357d.png)

@humerh and @mattia-leone can you please have a look at the current results produced by Emikat for the Adaptation options calculation? It seems that right now, applying our adaptation strategies only make things worse in the end.

**Comment by patrickkaleta** [comment]
@mattia-leone and @stefanon **FYI:**
Heinrich fixed an error in the calculations. I re-triggered the calculations and the results now seem to be better, however the adapted scenario still shows sligthly higher temperatures than the baseline scenario.
BTW: Feel free to run your own adapted scenarios for your own Studies. Remember that in the development instance you cannot trigger the Emikat calculations. You can however later use the development instance to view the results if you like.

**For comparison:**
[Table for baseline scenario temperatures ](https://csis.myclimateservice.eu/study/25/step/3682/view/table)
[Table for adapted scenario temperatures](https://csis-dev.myclimateservice.eu/study/25/step/3682/view/table)

As can be seen in the tables, T_MRT, T_UTCI and T_A are on average slightly higher in the adapted scenario.

**Comment by DenoBeno** [comment]
@mattia-leone has told me today that they have found some error in calculation. Didn't get it exactly, but it could be that we are multiplying the values of old parameters with new ones instead of replacing them.

**Comment by mattia-leone** [comment]
Hi @patrickkaleta , did you check Martina's email? It relates with what @DenoBeno says. Actually is the opposite, before we put "multiply value" for the Ts parameter, which is wrong (and that could explain the higher values in the adaptation), while now Martina corrected this with "replace value"
Hopefully if you run another calculation we might be happy now..
Please let us know

P.S. working on the impact stuff, hope to send what agreed in the telco over the weekend

**Comment by patrickkaleta** [comment]
> Hi @patrickkaleta , did you check Martina's email? It relates with what @DenoBeno says. Actually is the opposite, before we put "multiply value" for the Ts parameter, which is wrong (and that could explain the higher values in the adaptation), while now Martina corrected this with "replace value"
> Hopefully if you run another calculation we might be happy now..
> Please let us know
> 
> P.S. working on the impact stuff, hope to send what agreed in the telco over the weekend

Hi @mattia-leone, yes I've read her mail. We will deploy her changes to the live CSIS soon (I'm assuming today or tomorrow) and then we can see how that affects the calculations for the adaptation options.

**Comment by patrickkaleta** [comment]
> Hi @patrickkaleta , did you check Martina's email? It relates with what @DenoBeno says. Actually is the opposite, before we put "multiply value" for the Ts parameter, which is wrong (and that could explain the higher values in the adaptation), while now Martina corrected this with "replace value"
> Hopefully if you run another calculation we might be happy now..
> Please let us know
> 
> P.S. working on the impact stuff, hope to send what agreed in the telco over the weekend

The changes made by Martina have been synchronized onto our PROD system and I triggered the Emikat calculation again. Looking at the results, I don't see any changes. However, I noticed that the [AO Effects taxonomy](https://csis-dev.myclimateservice.eu/admin/structure/taxonomy/manage/localeffectsparameters/overview) has some problems. Some effects are included multiple times: 
![AO-effects-duplicated](https://user-images.githubusercontent.com/42167036/86779227-9b670b80-c05b-11ea-82ba-759f89761790.png)

I double-checked with the DEV instance, to make sure that those duplicates weren't caused by a faulty synchronization into PROD, but these duplicates already existed on the DEV server before. I didn't check all of them, but for instance the Surface temperature = 1.27 effect is in the system twice and [one](https://csis-dev.myclimateservice.eu/taxonomy/term/5712/edit?destination=/admin/structure/taxonomy/manage/localeffectsparameters/overview) of them still uses "multiply" instead of "replace value". Therefore, it's possible that Emikat is still supplied with wrong parameters, which lead to wrong calculation results.

@mattia-leone and @martinapizzicato please check again all the AO effects and remove unnecessary duplicates. An AO effect only needs to be included once in this taxonomy and then it can be referenced multiple times in different AOs. And make sure that the AOs then reference only the remaining (and hopefully correct) ones.

**Comment by martinapizzicato** [comment]
> > Hi @patrickkaleta , did you check Martina's email? It relates with what @DenoBeno says. Actually is the opposite, before we put "multiply value" for the Ts parameter, which is wrong (and that could explain the higher values in the adaptation), while now Martina corrected this with "replace value"
> > Hopefully if you run another calculation we might be happy now..
> > Please let us know
> > P.S. working on the impact stuff, hope to send what agreed in the telco over the weekend
> 
> The changes made by Martina have been synchronized onto our PROD system and I triggered the Emikat calculation again. Looking at the results, I don't see any changes. However, I noticed that the [AO Effects taxonomy](https://csis-dev.myclimateservice.eu/admin/structure/taxonomy/manage/localeffectsparameters/overview) has some problems. Some effects are included multiple times:
> ![AO-effects-duplicated](https://user-images.githubusercontent.com/42167036/86779227-9b670b80-c05b-11ea-82ba-759f89761790.png)
> 
> I double-checked with the DEV instance, to make sure that those duplicates weren't caused by a faulty synchronization into PROD, but these duplicates already existed on the DEV server before. I didn't check all of them, but for instance the Surface temperature = 1.27 effect is in the system twice and [one](https://csis-dev.myclimateservice.eu/taxonomy/term/5712/edit?destination=/admin/structure/taxonomy/manage/localeffectsparameters/overview) of them still uses "multiply" instead of "replace value". Therefore, it's possible that Emikat is still supplied with wrong parameters, which lead to wrong calculation results.
> 
> @mattia-leone and @martinapizzicato please check again all the AO effects and remove unnecessary duplicates. An AO effect only needs to be included once in this taxonomy and then it can be referenced multiple times in different AOs. And make sure that the AOs then reference only the remaining (and hopefully correct) ones.

Hi @patrickkaleta , could you give me the authorization to delete them? I'm using Mattia Leone's account.
I don't know why, but sometimes the csis duplicates my uploads. It also happened with some showcases, i noticed it with double hazard showcases (like heat and flood for Bo01 project).

**Comment by patrickkaleta** [comment]
> Hi @patrickkaleta , could you give me the authorization to delete them? I'm using Mattia Leone's account.

You should now be able to remove terms from the [LocalEffectsParameters](https://csis-dev.myclimateservice.eu/admin/structure/taxonomy/manage/localeffectsparameters/overview) taxonomy on the DEV instance.

> I don't know why, but sometimes the csis duplicates my uploads. It also happened with some showcases, i noticed it with double hazard showcases (like heat and flood for Bo01 project).

That is a strange and unique behaviour. Can't really help without further information. Are you maybe using the "go back" button of your browser after you clicked on the Save button down at the bottom of the form? In that case you would come back to the form (with already filled out information from the previous item) and if you click the Save button again, that item will be then created a second time.

**Comment by patrickkaleta** [comment]
@mattia-leone @stefanon @martinapizzicato all changes and updates made to the Adaptation options are available on our PROD instance of the CSIS, but Emikat still returns the same results as before. At least in the Table the results haven't changed. I can't currently see any results on the Map. Maybe an issue with Emikat's GeoServer @humerh ?

**Comment by patrickkaleta** [comment]
I checked a couple of the Adaptation options in the CSIS and compared the parameters with the Excel sheet provided by @martinapizzicato and I found in roughly half of them either missing or wrong parameters. An extreme example was the _Removable canony Type B_ Adaptation option with wrong **albedo, emissivity, runoff and transmissivity.**
![wrong-parameters-2](https://user-images.githubusercontent.com/42167036/88038534-a2574900-cb46-11ea-8247-10098b65b0f4.png)

No surprise that we are not getting any valid results with this. I changed the used Adaptation Project for Study 25 to only contain a single strategy for _Build open spaces_ with AOs that I've now checked and corrected. Also, I set the application coefficient to 100%, so all land-use categories applicable to this strategy should be affected (that should give us more deviating results from the baseline scenario). 

Let's see if we can get some realistic results now. @mattia-leone @stefanon and @martinapizzicato please double-check the adaptation options in the CSIS DEV-instance to make sure that all the parameters are set correctly. I'm not ruling out the possibility of calculation errors inside Emikat, but as long as the parameters are not all 100% correct, it will be difficult for @humerh to check the calculations made by Emikat.

**Comment by martinapizzicato** [comment]
@patrickkaleta i'm so sorry, maybe i've sent you the old excell file. The right parameters are the ones I loaded on the csis. I'll 
immediately send you the correct file.

**Comment by patrickkaleta** [comment]
> @patrickkaleta i'm so sorry, maybe i've sent you the old excell file. The right parameters are the ones I loaded on the csis. I'll
> immediately send you the correct file.

Ok thanks for sending the updated version. But even in this new Excel sheet e.g. the albedo for  Trees Type B - Medium Canopy diameter is set to be 0.2, while in the CSIS it was set to 0.6 (I've already corrected that parameter to test the calculation in Emikat), so we need to double-check all the parameters in CSIS to ensure that Emikat is getting the correct values.

**Comment by martinapizzicato** [comment]
![ALBEDO](https://user-images.githubusercontent.com/64830372/88046787-e3ecf180-cb50-11ea-984a-b143c4afd6f9.jpg)

I don't know why, but maybe you don't see my changes on CSIS (as you can see in the image below they are identical)

**Comment by patrickkaleta** [comment]
I confused "Trees Type B - Medium canopy" with "Canopy B - Removable Canopy". I was actually talking about "Canopy B - Removable Canopy". For that AO the albedo was set to 0.6 in the CSIS, when it was actually supposed to be set to 0.2. It is now showing the correct value in the CSIS, because I fixed it this morning, while I was reviewing some of the available AOs.

**Comment by patrickkaleta** [comment]
I checked in total maybe around 6-8 AOs and found a couple of parameters that did not match their expected values based on the provided Excel sheet. I fixed the parameters for those but I don't have the time to now check all of the available AOs and their parameters, so that is something that PLENIVS needs to take care of now.

**Comment by martinapizzicato** [comment]
@patrickkaleta it's very strange, because I already re-checked the parameters on https://csis-dev.myclimateservice.eu/ and they are the same as the excell. 
Which A0 do you see different from those of the file? 
If the problem persists we can have a call this afternoon, also with @mattia-leone

**Comment by p-a-s-c-a-l** [comment]
Are you changing the adaption options on CSIS or CSIS-DEV?

**Comment by martinapizzicato** [comment]
@p-a-s-c-a-l CSIS-DEV, as I have been told

**Comment by p-a-s-c-a-l** [comment]
O.K. Apparently @patrickkaleta updated the AOs on [CSIS-DEV](https://csis-dev.myclimateservice.eu/admin/content?title=&type=adaptation_options&status=All&langcode=All&order=changed&sort=desc) and [CSIS PROD](https://csis.myclimateservice.eu/admin/content?title=&type=adaptation_options&status=All&langcode=All&order=changed&sort=desc).

**Comment by patrickkaleta** [comment]
> O.K. Apparently @patrickkaleta updated the AOs on [CSIS-DEV](https://csis-dev.myclimateservice.eu/admin/content?title=&type=adaptation_options&status=All&langcode=All&order=changed&sort=desc) and [CSIS PROD](https://csis.myclimateservice.eu/admin/content?title=&type=adaptation_options&status=All&langcode=All&order=changed&sort=desc).

Yes, I updated the AOs on CSIS-DEV and then synchronized the AOs over to PROD.

**Comment by patrickkaleta** [comment]
> Which A0 do you see different from those of the file?

As I already mentioned, it was the [Shading Systems - Removable canopy (Type B)](https://csis.myclimateservice.eu/node/569), which was showing different parameters than the Excel sheet (see screenshot from my previous [comment](https://github.com/clarity-h2020/csis/issues/154#issuecomment-661754280) ). I've than fixed that AO (as well as a couple of others), so now these are showing the correct parameters. But I haven't had the time to check all the others, so that is something that needs to be done now.

**Comment by martinapizzicato** [comment]
![removable](https://user-images.githubusercontent.com/64830372/88050251-c3c03100-cb56-11ea-897d-4dc46a0628d6.jpg)

@patrickkaleta I keep seeing the same parameters

**Comment by martinapizzicato** [comment]
@patrickkaleta I checked all the other AOs parameters again, and the values match the excell sheet.

**Comment by patrickkaleta** [comment]
> ![removable](https://user-images.githubusercontent.com/64830372/88050251-c3c03100-cb56-11ea-897d-4dc46a0628d6.jpg)
> 
> @patrickkaleta I keep seeing the same parameters

Yes, because you are now already looking at the corrected version of this AO. This morning the albedo was set to 0.6 and I then fixed that so it now says 0.2. On both instances, DEV and PROD, this AO is now showing the correct value in accordance to what is given in the Excel sheet.

**Comment by patrickkaleta** [comment]
> @patrickkaleta I checked all the other AOs parameters again, and the values match the excell sheet.

Ok, that's great news. I was probably just unlucky (or lucky) that I stumbled upon an AO with wrong parameters this morning.

**Comment by martinapizzicato** [comment]
@patrickkaleta maybe it was my mistake, i'm sorry. The important thing is that now all the parameters are right. Let me know if you have any other problems!

**Comment by patrickkaleta** [comment]
> @patrickkaleta maybe it was my mistake, i'm sorry. The important thing is that now all the parameters are right. Let me know if you have any other problems!

No problem, there are a lot of different parameters and it's not easy to keep track of  all of them. But I'm still not sure that everything is ok now:

![green-pergolas](https://user-images.githubusercontent.com/42167036/88051718-5792fc80-cb59-11ea-9357-f383df73a664.png)

According to the new Excel sheet the Adaptation option [Green pergolas Type C ](https://csis-dev.myclimateservice.eu/node/571) is supposed to have a Transmissivity parameter, but in the CSIS this parameter is missing.

**Comment by patrickkaleta** [comment]
...and that was literally the first AO I randomly clicked on and it had a missing parameter. So I wouldn't be surprised if other AOs are also having some missing or wrong parameters.

**Comment by patrickkaleta** [comment]
Until we are sure that the AOs in the CSIS have all correct parameters, I will temporarilly create a dummy AO directly on PROD, which will only change the Albedo parameter and which we can use to verify the basic functionality of the Emikat calculations. I will pick a very low albedo and apply it to 100% of the area - that way we should definitely get lower temperatures as a result. If that's not the case, that we know that we have to adjust something in the Emikat calculations as well.

**Comment by martinapizzicato** [comment]
> > @patrickkaleta maybe it was my mistake, i'm sorry. The important thing is that now all the parameters are right. Let me know if you have any other problems!
> 
> No problem, there are a lot of different parameters and it's not easy to keep track of all of them. But I'm still not sure that everything is ok now:
> 
> ![green-pergolas](https://user-images.githubusercontent.com/42167036/88051718-5792fc80-cb59-11ea-9357-f383df73a664.png)
> 
> According to the new Excel sheet the Adaptation option [Green pergolas Type C ](https://csis-dev.myclimateservice.eu/node/571) is supposed to have a Transmissivity parameter, but in the CSIS this parameter is missing.

@patrickkaleta fixed it

**Comment by patrickkaleta** [comment]
> @patrickkaleta fixed it

Just this one Adaptation option or are all AOs now double-checked and corrected?

**Comment by martinapizzicato** [comment]
@patrickkaleta all AOs are corrected

**Comment by NICDDB** [comment]
Dear all,  as mattia said in the last dev call, we did the same calculation of the adaptation with our local model. 
These are the results on some sample cells, showing the improvements in terms of tmrt and utci:

Default landuse (no adaptation option)
![landuse stato di fatto](https://user-images.githubusercontent.com/60875355/88051389-d0de1f80-cb58-11ea-845c-4f5d95673f0e.png)

landuse with adaptation option
![landuse adaptation](https://user-images.githubusercontent.com/60875355/88051558-1b5f9c00-cb59-11ea-91ad-ae844375b698.png)

Tmrt default landuse
![tmrt stat di fatto](https://user-images.githubusercontent.com/60875355/88051396-d3407980-cb58-11ea-8130-865a18198090.png)

Tmrt landuse with adaptation
![tmrt post-adaptation](https://user-images.githubusercontent.com/60875355/88051395-d2a7e300-cb58-11ea-9db6-609e88306a26.png)

UTCI default landuse
![utci stat di fatto](https://user-images.githubusercontent.com/60875355/88051400-d3d91000-cb58-11ea-98bf-4ded10a00ed9.png)

UTCI with adaptation
![UTCI post-adaptation](https://user-images.githubusercontent.com/60875355/88051398-d3d91000-cb58-11ea-8738-afec1fcc5e2e.png)

Following the table with parameters used for current state landuse and for adaptation solutions:

current state layers of landuse

classref | class | albedo | emissivity | transmissi | vegeshadow | runoffcoef | hillsha_gf | new_hillshadeb
-- | -- | -- | -- | -- | -- | -- | -- | --
1 | agricultural areas | 0.16 | 0.94 | 1.0 | 1 | 0.1 | 0.97 | 0.8/0.9
3 | buildings | 0.18 | 0.95 | 1.0 | 1 | 0.9 | 0.97 | 1.0
4 | Built open spaces | 0.25 | 0.9 | 1.0 | 1 | 0.75 | 0.97 | 0.8/0.9
6 | roads | 0.18 | 0.95 | 1.0 | 1 | 0.9 | 0.97 | 0.8/0.9
8 | trees | 0.13 | 0.97 | 0.25 | 0 | 0.05 | 0.37 | 1.0
9 | vegetated areas | 0.21 | 0.94 | 0.25 | 1 | 0.18 | 0.97 | 0.8/0.9



adaptation solution layers

classref | class | albedo | emissivity | transmissi | vegeshadow | runoffcoef | hillsha_gf | new_hillshadeb
-- | -- | -- | -- | -- | -- | -- | -- | --
11 | green surface a | 0.16 | 0.97 | 1.0 | 1 | 0.15 | 0.97 | 0.8/0.9
12 | green surface b | 0.2 | 0.95 | 1.0 | 1 | 0.18 | 0.97 | 0.8/0.9
13 | green surface c | 0.25 | 0.96 | 1.0 | 1 | 0.22 | 0.97 | 0.8/0.9
14 | trees a | 0.2 | 0.98 | 0.25 | 0 | 0.05 | 0.25 | 0.92
15 | trees b | 0.2 | 0.97 | 0.25 | 0 | 0.05 | 0.4 | 0.92
16 | trees c | 0.2 | 0.96 | 0.25 | 0 | 0.05 | 0.51 | 0.92
17 | permeable floring a | 0.35 | 0.95 | 1.0 | 1 | 0.35 | 0.97 | 0.8/0.9
18 | permeable floring b | 0.4 | 0.93 | 1.0 | 1 | 0.45 | 0.97 | 0.8/0.9
19 | permeable floring c | 0.6 | 0.85 | 1.0 | 1 | 0.6 | 0.97 | 0.8/0.9
21 | reflective surface b | 0.5 | 0.8 | 1.0 | 1 | 0.75 | 0.97 | 0.8/0.9
24 | green roofs b | 0.26 | 0.96 | 1.0 | 1 | 0.35 | 0.97 | 1.0
26 | cool roofs b | 0.66 | 0.9 | 1.0 | 1 | 0.75 | 0.97 | 1.0
28 | canopy a | 0.6 | 0.92 | 0.25 | 1 | 0.75 | 0.97 | 0.2
29 | canopy b | 0.6 | 0.95 | 0.25 | 1 | 0.75 | 0.97 | 0.4
30 | canopy c | 0.2 | 0.96 | 0.25 | 1 | 0.75 | 0.6 | 0.92
32 | water bodies b | 0.69 | 0.97 | 1.0 | 1 | 0.85 | 0.97 | 0.8/0.9

N.B. the parameter new_hillshadeb (hillshade building) depend by built density. In our case the density was low or medium so hillshade building was 0.8 or  0.9 for the landuse that are not buildings, trees or canopy.

**Comment by p-a-s-c-a-l** [comment]
**FYI:** I've updated HC-LE and IA tables to include all columns and adaptation scenarios when **downloading** the tables as JSON or CSV, so you can now compare adapted vs non-adapted scenarios offline. Map Comparison for IA is also [available](https://github.com/clarity-h2020/csis/issues/177#issuecomment-663020432).

I had some problems calculating adaptation scenarios, but at least [this test study](https://csis-dev.myclimateservice.eu/study/163/step/4891/view/table) contains some data.

**Comment by p-a-s-c-a-l** [comment]
**FYI:** You can know compare Baseline vs adaptation in Scenario Analysis (currently only in DEV-CSIS). See https://github.com/clarity-h2020/csis/issues/177#issuecomment-663539634 for more details.

It seems that HW impact with adaptation is higher than without.

**Comment by humerh** [comment]
From my point of view I found no Error in calculation implementation.

For the AO "Extensive green roof 100%" the used paramters are:

**Comment by humerh** [comment]
For example for Layer 9 (DENSE URBAN FABRIC):

T_SURFACE_DELTA:   10 -> 1.38
EMISSIVITY_SURFACE: 0.9 -> 0.96
ALBEDO: 0.25 -> 0.26
TRANSMISSIVITY  0.02  -> 1

Are this realy GOOD values???

**Comment by patrickkaleta** [comment]
For Study 25 I'm currently running the re-calculation with a dumm AS, which features only the albedo parameter set to a very high value (0.86) and is set to be used on 100% of all buildings. Once the calculation is done, we can compare the results between baseline and adapted scenario.

**Comment by NICDDB** [comment]
Yes @humerh these values are good. 

I have just one doubt: for T_SURFACE_DELTA you change value from 10 to 1.38. I don't understand the start value (10), is maybe just a typing mistake?

**Comment by patrickkaleta** [comment]
> For [Study 25](https://csis.myclimateservice.eu/study/25/view/summary) I'm currently running the re-calculation with a dummy AS, which features only the albedo parameter set to a very high value (0.86) and is set to be used on 100% of all buildings. Once the calculation is done, we can compare the results between baseline and adapted scenario.

Calculations are completed. Looking at the results in the [HC-LE table](https://csis.myclimateservice.eu/study/25/step/3682/view/table) the temperatures stayed mostly the same or in case of T_MRT are sometimes higher in the adapted scenario by 0,01°C. **That seems wrong.**

The adaptation strategy is set to be applied to everything that counts as buildings, which is definitely the case for the selected Study area and therefore I would expect the temperatures to go down more drastically for a lot of cells.
![buildings-inside-study-area](https://user-images.githubusercontent.com/42167036/88530887-54d55300-d002-11ea-8704-fa54f0a1a553.png)

**Comment by DenoBeno** [comment]
Following today's discussion, I have implemented the following: 

1. added an "AO Implementation" step. This step should be similar to Risk/Impact assessment, but allow comparison of the situation with and without AOs. Template:  https://csis-dev.myclimateservice.eu/node/5105/edit/template
2. Updated the AO appraisal step so that no data is shown in it.
3. Shortened the labels for all other steps. (see https://csis-dev.myclimateservice.eu/taxonomy/term/2217/edit)

The resulting study step labels look like this now:
![image](https://user-images.githubusercontent.com/2961790/88538213-3fb2f100-d00f-11ea-8bca-9c10df178ec0.png)

@p-a-s-c-a-l : I am not sure which elements should be integrated int eh last step. Please check and edit the template accordingly.

**Comment by humerh** [comment]
If I feed the formula for T_MRT with the same parameters and only increase the ALBEDO, I get always and increased FLUX and T_MRT.

If I study the formula for fluxes and T_MRT, which I harvested from PLINIUS, this behavior is also understandable.

Some typical parts of the formula:

    Alfa  FLOAT := albedo;   -- albedo, considered constant 0.15
    k_in := k_in + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    k_out := k_in * Alfa;

So these terms always influence the the result in a direct proportional way ... the hight Albedo, the higher the flux.

@stephanon or @mattia-leone , can you please check the formula and give feedback.

**Comment by humerh** [comment]
Defined FUNCTION in SQL

 create or replace FUNCTION           hw_localeffect_fluxes (
    eta_angle     FLOAT,
    theta_angle   FLOAT,
    t_air         FLOAT,
    t_s           FLOAT,
    gswrad        FLOAT,
    directrad     FLOAT,
    diffuserad    FLOAT,
    albedo        FLOAT,
    emissivity    FLOAT,
    transmissi    FLOAT,
    vegeshadow    SMALLINT,
    buildingsh    SMALLINT,
    hillsha_gf    FLOAT,
    hillshadeb    FLOAT,
    build_dens    INTEGER)
    RETURN FLOAT
AS
    --
    --  A simplified model for Tmrt evaluation in urban areas,
    --  mainly developed after the following papers:
    -- F.Lindberg and al. - SOLWEIG 1.0 - Modeling spatial variations of 3D radiant fluxes and mean radiant temperature in complex urban settings
    --  F.Lindberg and al. - The influence of vegetation and building morphology on shadow patterns and mean radiant temperatures in urban areas: model development and evaluation
    --  and Alessandra Capolupo first formulation in the excel model sheets
    --
    -- Please, note this main assumptions for Tmrt calculation with this simplified procedure:
    --
    -- completely clear sky
    -- valid daytime only
    --  if not available Ts can be estimated from Ta using an increment factor
    --  and passed as an input parameter (test values deduce by analyzing SOLWEIG simulations results, done only on Naples area and summer time)
    --
    -- ...
    --  [stefanon nardone - PLINIVS - LUPT]

    -- input parameters:
    --
    -- eta_angle double precision  Sun's altitude angle (degrees)
    -- theta_angle double precision   Sun's azimuth angle (degrees)
    -- t_air double precision    Air temperature (°C)
    -- t_s double precision    Soil temperatue (°C)
    -- gswrad double precision   Global shortwave radiation from https://meteoexploration.com/products/SolarCalculator.html (Wh/m2)
    -- gswrad double precision   Global shortwave radiation from https://meteoexploration.com/products/SolarCalculator.html (Wh/m2)
    -- albedo double precision
    -- emissivity double precision
    -- transmissi double precision
    -- vegeshadow smallint
    -- buildingsh smallint
    -- hillsha_gf double precision
    -- hillshadeb double precision
    -- build_dens integer

    -- costants
    ST_BOLTZ   CONSTANT FLOAT := 5.67E-8;         -- stefan-boltzmann constant
    E_P        CONSTANT FLOAT := 0.97;                      -- body emissivity
    XI_K       CONSTANT FLOAT := 0.7; -- absorption coefficient for shortwave radiation
    F_AB       CONSTANT FLOAT := 0.06; -- Fi angular factor for above and below
    F_4P       CONSTANT FLOAT := 0.22;             -- Fi for 4 cardinal points
    --
    G          CONSTANT FLOAT := gswrad; -- 942.37; Global shortwave radiation from https://meteoexploration.com/products/SolarCalculator.html (Wh/m2)
    D          CONSTANT FLOAT := diffuserad;  --0.3 * gswrad; -- modificato da 0.1 a 0.3 -- Diffuse shortwave radiation
    I          CONSTANT FLOAT := directrad;  -- 0.7 * gswrad; -- modificato da 0.9 a 0.7 -- Direct shortwave radiation
    --
    --
    ETA                 FLOAT := eta_angle/180*3.14159; --  sun's altitude angle above the horizon. max elevation for Naples @21/6
    thetaE               FLOAT := theta_angle/180*3.14159; --  sun's azimuth angle init to position for Naples @21/6 @max elevation time

    TA                  FLOAT := (t_air) + 273.15; -- air temperature (converted to °K)
    TS                  FLOAT := t_s + 273.15; -- surface temperature (converted to °K)
    --
    E_SKY               FLOAT := 0.787 + 0.0028 * (TA - 273.15); -- sky emissivity [Ta is in C]

    -- E_W   float8 :=  0.7;       --
    -- ^^ modificato diventa funzione della densità dell'edificato ..
    --
    TYPE FLOATARR4 IS VARRAY (4) OF FLOAT;

    E_W                 FLOATARR4
                            := FLOATARR4 (0.01,
                                          0.1,
                                          0.4,
                                          0.7); -- E_W values attributed as a function of building densities classes [0..3] low - high
    --
    -- aliases
    Es                  FLOAT := emissivity;
    Sv                  SMALLINT := vegeshadow; -- vegetation shadow (boolean)
    Sb                  SMALLINT := buildingsh;   -- building shadov (boolean)
    Tau                 FLOAT := transmissi;
    Psi_b               FLOAT := hillshadeb;                                --
    Psi_v               FLOAT := hillsha_gf;                                --
    Alfa                FLOAT := albedo;   -- albedo, considered constant 0.15
    --
    -- variables
    tmrt                FLOAT;                     -- mean radiant temperature
    mrfd                FLOAT;                -- (R) mean radiant flux density
    k_in                FLOAT;           -- K shortwave radiation flux (W/m^2)
    k_out               FLOAT;                                              --
    k_l                 FLOATARR4 := FLOATARR4(0,0,0,0);                                          --
    l_in                FLOAT;            -- L longwave radiation flux (W/m^2)
    l_out               FLOAT;                                              --
    l_l                 FLOATARR4 := FLOATARR4(0,0,0,0);                                           --
    --
    --theta                FLOAT;
    theta_e              FLOAT;
    theta_s              FLOAT;
    theta_w              FLOAT;
    theta_n              FLOAT;                -- azimuth E, S, W, N components
    --cnt               INTEGER;                               -- loop counter
BEGIN
    -- Kin
    k_in := (I * (Sb - (1 - Sv) * (1 - Tau)) * SIN (ETA)); -- angles are in degrees
    k_in := k_in + D * (Psi_b - (1 - Psi_v) * (1 - Tau));
    k_in := k_in + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    --                     ^^^ 0.5 is another parameter for evaluate (1-Fs) factor in the formula, with Fs fraction of building walls that is shadowed

    -- Kout
    k_out := k_in * Alfa;
    --
    theta_e := thetaE;
    theta_s := thetaE - 3.14159/2;   -- 90° 
    theta_w := thetaE - 3.14159;     -- 180°
    theta_n := thetaE - 3/2*3.14159; -- 270°

    --theta := theta_angle;

    -- k_l[1] = East -- easterly component
    IF theta_angle > 0.0 OR theta_angle <= 180.0
    THEN
        k_l (1) :=
              I * (Sb - (1 - Sv) * (1 - Tau)) * COS (ETA) * SIN (theta_e)
            + D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    --       ^^^ 0.5 = (1-Fs) : Fs fraction of wall shadowed ...
    ELSE
        k_l (1) :=
              D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa *  0.5* (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    END IF;

    -- k_l[2] = South -- southerly component
    IF theta_angle > 90.0 AND theta_angle <= 270.0
    THEN
        k_l (2) :=
              I * (Sb - (1 - Sv) * (1 - Tau)) * COS (ETA) * SIN (theta_s)
            + D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    ELSE
        k_l (2) :=
              D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    END IF;

    -- k_l[3] = West
    IF theta_angle > 180.0 AND theta_angle <= 360.0
    THEN
        k_l (3) :=
              I * (Sb - (1 - Sv) * (1 - Tau)) * COS (ETA) * SIN (theta_w)
            + D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    ELSE
        k_l (3) :=
              D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    END IF;

    -- k_l[4] = North -- northerly component
    IF theta_angle <= 90.0 OR theta_angle > 270.0
    THEN
        k_l (4) :=
              I * (Sb - (1 - Sv) * (1 - Tau)) * COS (ETA) * SIN (theta_n)
            + D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    ELSE
        k_l (4) :=
              D * (Psi_b - (1 - Psi_v) * (1 - Tau))
            + G * Alfa * 0.5 * (1 - (Psi_b - (1 - Psi_v) * (1 - Tau)));
    END IF;

    -- Lin
    l_in :=
          (Psi_b + Psi_v - 1) * E_SKY * ST_BOLTZ * POWER (TA, 4)
        + (2 - Psi_v - Psi_b) * ST_BOLTZ * E_W (build_dens + 1) * POWER (TA, 4)
        + (Psi_v - Psi_b) * E_W (build_dens + 1) * E_SKY * ST_BOLTZ * POWER (TS, 4)
        + (2 - Psi_b - Psi_v) * (1 - E_W (build_dens + 1)) * ST_BOLTZ  * POWER (TA, 4) * E_SKY;

    -- Lout
    l_out := Es * ST_BOLTZ * POWER (TS + (TS - TA), 4);

    -- L_l
    FOR cnt IN 1 .. 4
    LOOP
        l_l (cnt) := 0.5 * l_out;             -- simplified lateral components
    END LOOP;

    -- mrfd
    mrfd := XI_K * k_in * F_AB + E_P * l_in * F_AB;                   -- Above
    mrfd := mrfd + XI_K * k_out * F_AB + E_P * l_out * F_AB;          -- Below

    FOR cnt IN 1 .. 4
    LOOP
        mrfd := mrfd + XI_K * k_l (cnt) * F_4P + E_P * l_l (cnt) * F_4P; -- 4 cardinal points
    END LOOP;

    -- Tmrt ( output in °C)
    tmrt := mrfd / (ST_BOLTZ * E_P);
    tmrt := POWER (tmrt,0.25) - 273.15;

    -- return fluxes
    RETURN mrfd;
 END hw_localeffect_fluxes;

**Comment by DenoBeno** [comment]
FYI, I have started making some help-views to aid with debugging this. First one is: https://csis-dev.myclimateservice.eu/maintenance/local-effects. 

Values defined for the T_SURFACE delta are definitely wrong, because some of them are with unit (%), some without unit and some are with "replace value" and other with "mutiply". In addition, some are definitely wrong, see screenshot:

![image](https://user-images.githubusercontent.com/2961790/88554877-55ccab80-d027-11ea-82ac-7d73556be838.png)

Moreover, I believe that even the titles are wrong.

Logically, the surface temperature should be multiplied by a factor (e.g.) 1.22 to make it 22% higher. However, I see that the actual parameter is called T_SURFACE_DELTA, so I presume that "replace" is a right operation. I will therefore bulk-edit all of them now to use "replace value" and have no unit. Because 1.22% is not 22%.

In addition, "Surface temperature = 1.07" is also incorrect. Should be "Surface temperature +7%" 

I'll fix all surface temperatures now so that we can test. Please double-check all other parameters.

**Comment by mattia-leone** [comment]
Martina said again that she fixed all, but apparently it's not.. in her view she doesn't see these values... maybe it's better you fix it now so that we finalize this steps
From this screenshot:
- First two rows should be erased
- All rows with multiply should be changed with replace value

**Comment by mattia-leone** [comment]
of course none of them should have % as unit

**Comment by humerh** [comment]
My main problem is:
- If we only encrease Albedo, then the formula has the behavior, that also the T_MRT encreases.
- Is this the result, which we expect?? 

Yesterday Mattia said, the behavior should be the other way.

**Comment by patrickkaleta** [comment]
In addition to what @DenoBeno added yesterday [showing all the LE parameters individually](https://csis-dev.myclimateservice.eu/maintenance/local-effects), I've now also updated the [Adaptation options overview](https://csis-dev.myclimateservice.eu/maintenance/adaptation-options) in our **Maintenance** tab:
![AO-overview](https://user-images.githubusercontent.com/42167036/88639096-ae9a5380-d0bc-11ea-9989-4fe9e51ecd6b.png)

By clicking on the individual AOs you'll get to a more detailed view, where you can see the used LE parameters and the Adaptation Strategies, where this AO is included. If you see a wrong LE parameter, you can directly edit it from there.
![AO-detail](https://user-images.githubusercontent.com/42167036/88639944-a4c52000-d0bd-11ea-8e98-3607a10f4a98.png)

**Comment by mattia-leone** [comment]
> My main problem is:
> 
> * If we only encrease Albedo, then the formula has the behavior, that also the T_MRT encreases.
> * Is this the result, which we expect??
> 
> Yesterday Mattia said, the behavior should be the other way.

Dear @humerh you will get a direct feedback from @stefanon asap, so to explain in detail the "expected effect" of the main parameters on the Tmrt. 
Just to be more precise, indeed a higher albedo COULD result in a lower Tmrt BUT albedo is a very peculiar parameter, because in areas where building or vegetation shadow is low, a high albedo can also result in higher Tmrt, since all the radiation reflected on the surface hits back the people and increase the heat discomfort.
As said yesterday, the most "straightforward" parameter to test the effect of adaptation is the Ts, which for the adaptation measures is usually lower than existing land uses.
I am really sorry about this prolonged issue on the revision of the adaptation parameters on the CSIS, I hope we are on the right track now.

**Comment by patrickkaleta** [comment]
> ... maybe it's better you fix it now so that we finalize this steps

@mattia-leone 
I don't think @DenoBeno  has the time to do that and my remaining hours in this project are almost drained, so I can't do this either. Please use the pages that Denis and I created to get a better overview of how the parameters are currently configured in the CSIS and fix everything that is not right. **You will need to double-check, so this is not a one-person-job that can be done alone by @martinapizzicato.** 

> Martina said again that she fixed all, but apparently it's not.. in her view she doesn't see these values

What can't she see? Your user should have all the necessary permissions to be able to view the pages Denis and I created correctly.

**Comment by patrickkaleta** [comment]
> As said yesterday, the most "straightforward" parameter to test the effect of adaptation is the Ts, which for the adaptation measures is usually lower than existing land uses.

@mattia-leone 
By "Ts" you mean the Transmissivity parameter? We have currently 5 of those defined in the CSIS (**Transmissivity = 0.25, Transmissivity = 0.97, Transmissivity = 1** and **Transmissivity +10%**). I can change our dummy AS to use a Transmissivity parameter instead of the Albedo parameter, so that Heinrich can check the results of that configuration. 

Which value shoud I use for this? Transmissivity = 0.25 or Transmissivity = 1?

**Comment by mattia-leone** [comment]
by Ts I mean 

T_SURFACE_DELTA | Surface temperature

**Comment by mattia-leone** [comment]
T_SURFACE_DELTA is the only parameter which you can play with "dummy" adaptation, for all the others the effect on Tmrt is a complex combination of all parameters, and you CANNOT expect that by changing one random parameter you will get a reduction in Tmrt. In fact the adaptation strategies are a careful combination of the different adaptation options, designed to provide the desired Tmrt reduction.

**Comment by mattia-leone** [comment]
I have just personally checked the list here https://csis-dev.myclimateservice.eu/maintenance/local-effects
and corrected the remaining errors. 
I can confirm this values are all correct.

**Comment by patrickkaleta** [comment]
> T_SURFACE_DELTA is the only parameter which you can play with "dummy" adaptation, for all the others the effect on Tmrt is a complex combination of all parameters, and you CANNOT expect that by changing one random parameter you will get a reduction in Tmrt. In fact the adaptation strategies are a careful combination of the different adaptation options, designed to provide the desired Tmrt reduction.

Ok, I will change the dummy AO to include only the T_SURFACE_DELTA parameter.

**Comment by stefanon** [comment]
Hi Heinrich and all

> My main problem is:
> 
> * If we only encrease Albedo, then the formula has the behavior, that also the T_MRT encreases.
> * Is this the result, which we expect??

about this problem, yes, it's the expected result. I've checked once more the formulas used in our simplified implementation after the SOLWEIG model described in the article in the reference at the beginning of the script code and the following ones, and the albedo parameter is used in a way that increasing its values leads to a higher Tmrt value. There are references also about the fact that using high-reflectance pavement will reduce pavement surface temperature, but increasing pavement reflectance will increase the mean radiant temperature owing to the increased reflected radiation hitting the human body. Albedo is the amount of solar radiation reflected from an object or surface, usually expressed as a percentage.
 
> Yesterday Mattia said, the behavior should be the other way.

I'm not sure about this, anyway the albedo values are low for dark and high for bright surfaces, in the range 0 - 1 (snow albedo is about 0.8, asphalt 0.04), but using only albedo variations for dummy adaptation isn't good for Tmrt evaluation. Parameters that better influence the results are the soil temperature (Ts), and those linked to the radiation flux, so the shadowing and the 'sky view factor' parameters, psi_b and psi_v (named hillshadeb and hillsha_gf in the code).

**Comment by humerh** [comment]
Thank you for this clarification. This explanation is understandable for me.
So we don't have to modify the formula.
Thanks

**Comment by p-a-s-c-a-l** [comment]
> @p-a-s-c-a-l : I am not sure which elements should be integrated int eh last step. Please check and edit the template accordingly.

done

**Comment by patrickkaleta** [comment]
I noticed someone switched back to "real" AOs in [Study 25](https://csis.myclimateservice.eu/study/25/step/3686/view/table), so I tried another re-calculation of the adapted scenario but I see no differences between baseline and adapted scenario ATM.

HC-LE shows exactly the same results in the table:
![baseline-vs-adapted](https://user-images.githubusercontent.com/42167036/88778846-4d8c8180-d189-11ea-95b7-e7107e4ffae1.png)

and the map layers for "local effect mean radiant temperature" also appear to be identical:
- [baseline](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27BASELINE%27%20and%20TIME_PERIOD=%27Baseline%27%20AND%20EMISSIONS_SCENARIO=%27Baseline%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&version=1.1.1&layers=clarity%3Aview.2974&styles=T_MRT%0D%0A&format=image%2Fpng&transparent=true&tileSize=1536&info_format=application%2Fjson&width=608&height=496&srs=EPSG%3A3857&bbox=1818436.9029043436%2C6143052.870708901%2C1824246.1170540168%2C6147791.966462584)
- [adapted](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27ADAPTATION-01%27%20and%20TIME_PERIOD=%27Baseline%27%20AND%20EMISSIONS_SCENARIO=%27Baseline%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&version=1.1.1&layers=clarity%3Aview.2974&styles=T_MRT%0D%0A&format=image%2Fpng&transparent=true&tileSize=1536&info_format=application%2Fjson&width=608&height=496&srs=EPSG%3A3857&bbox=1818436.9029043436%2C6143052.870708901%2C1824246.1170540168%2C6147791.966462584)

Risk/Impact table shows also identical results.

@humerh has the adaptation options calculation been temporarily stopped for debugging purposes or should it be theoretically working?

**Comment by humerh** [comment]
No, the calculation was not stopped, but the kind of calculation of Surface temperature was changed at the last visit in Napoli.

As you see in my script the T_SURFACE_DELTA parameter is not used any more, Instead of using this parameter the temperature correction is only done with T_SURFACE_FACTOR.
Please use this parameter.

--    EMIKATSYS.HW_LOCALEFFECT_FLUXES(ev.ETA, ev.Theta, ev.AIR_TEMP, ev.AIR_TEMP+gl.T_SURFACE_DELTA, ev.G, ev.I, ev.D, 
     EMIKATSYS.HW_LOCALEFFECT_FLUXES(ev.ETA, ev.Theta, ev.AIR_TEMP, ev.AIR_TEMP*gl.T_SURFACE_FACTOR, ev.G, ev.I, ev.D,

**Comment by patrickkaleta** [comment]
> No, the calculation was not stopped, but the kind of calculation of Surface temperature was changed at the last visit in Napoli.
> 
> As you see in my script the T_SURFACE_DELTA parameter is not used any more, Instead of using this parameter the temperature correction is only done with T_SURFACE_FACTOR.
> Please use this parameter.
> 
> -- EMIKATSYS.HW_LOCALEFFECT_FLUXES(ev.ETA, ev.Theta, ev.AIR_TEMP, ev.AIR_TEMP+gl.T_SURFACE_DELTA, ev.G, ev.I, ev.D,
> EMIKATSYS.HW_LOCALEFFECT_FLUXES(ev.ETA, ev.Theta, ev.AIR_TEMP, ev.AIR_TEMP*gl.T_SURFACE_FACTOR, ev.G, ev.I, ev.D,

@stefanon and @mattia-leone: 
In the CSIS all our surface temperature parameters are tagged as `T_SURFACE_DELTA`, which according to Heinrich is no longer used. Should that be changed to make all these parameters labeled as `T_SURFACE_FACTOR`?

The current parameters have values like 1.25, 1.30, 1.44,... which to me looks indeed more like a factor than a delta.

**Comment by stefanon** [comment]
@patrickkaleta what is used in the calculation for estimate Ts (T soil) from Ta (T air) is a multiplying factor, as in Ts = T_SURFACE_DELTA * Ta , where the name T_SURFACE_DELTA is confusing and linked to past versions.

**Comment by patrickkaleta** [comment]
> @patrickkaleta what is used in the calculation for estimate Ts (T soil) from Ta (T air) is a multiplying factor, as in Ts = T_SURFACE_DELTA * Ta , where the name T_SURFACE_DELTA is confusing and linked to past versions.

Ok I see, so I will rename `T_SURFACE_DELTA` to `T_SURFACE_FACTOR`, since it's a more fitting description and Emikat isn't recognizing `T_SURFACE_DELTA` anymore anyway.

**Comment by patrickkaleta** [comment]
After fixing the parameter name, it seems that we are finally getting some good results! Looking at [Study 25](https://csis.myclimateservice.eu/study/25/step/3682/view/table), T-MRT now goes down up to 3 °C in some of the cells:
![adapt-opt-calc](https://user-images.githubusercontent.com/42167036/89178430-da6e7b00-d58d-11ea-8618-7005ebae5845.png)

Resulting map layers look also promising:
- [Adapted Scenario](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27ADAPTATION-01%27%20and%20TIME_PERIOD=%2720710101-21001231%27%20AND%20EMISSIONS_SCENARIO=%27rcp85%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&version=1.1.1&layers=clarity%3Aview.2974&styles=T_UTCI%0D%0A&format=image%2Fpng&transparent=true&tileSize=1536&info_format=application%2Fjson&width=608&height=496&srs=EPSG%3A3857&bbox=1818436.9029043436%2C6143052.870708901%2C1824246.1170540168%2C6147791.966462584)
- [Baseline](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27BASELINE%27%20and%20TIME_PERIOD=%2720710101-21001231%27%20AND%20EMISSIONS_SCENARIO=%27rcp85%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&version=1.1.1&layers=clarity%3Aview.2974&styles=T_UTCI%0D%0A&format=image%2Fpng&transparent=true&tileSize=1536&info_format=application%2Fjson&width=608&height=496&srs=EPSG%3A3857&bbox=1818436.9029043436%2C6143052.870708901%2C1824246.1170540168%2C6147791.966462584)

**Comment by patrickkaleta** [comment]
> After fixing the parameter name, it seems that we are finally getting some good results! Looking at [Study 25](https://csis.myclimateservice.eu/study/25/step/3682/view/table), T-MRT now goes down up to 3 °C in some of the cells:
> ![adapt-opt-calc](https://user-images.githubusercontent.com/42167036/89178430-da6e7b00-d58d-11ea-8618-7005ebae5845.png)
> 
> Resulting map layers look also promising:
> 
> * [Adapted Scenario](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27ADAPTATION-01%27%20and%20TIME_PERIOD=%2720710101-21001231%27%20AND%20EMISSIONS_SCENARIO=%27rcp85%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&version=1.1.1&layers=clarity%3Aview.2974&styles=T_UTCI%0D%0A&format=image%2Fpng&transparent=true&tileSize=1536&info_format=application%2Fjson&width=608&height=496&srs=EPSG%3A3857&bbox=1818436.9029043436%2C6143052.870708901%2C1824246.1170540168%2C6147791.966462584)
> * [Baseline](https://service.emikat.at/geoserver/clarity/wms?CQL_FILTER=STUDY_VARIANT=%27BASELINE%27%20and%20TIME_PERIOD=%2720710101-21001231%27%20AND%20EMISSIONS_SCENARIO=%27rcp85%27%20AND%20EVENT_FREQUENCY=%27Rare%27%20AND%20SZ_ID=3183&&service=WMS&request=GetMap&version=1.1.1&layers=clarity%3Aview.2974&styles=T_UTCI%0D%0A&format=image%2Fpng&transparent=true&tileSize=1536&info_format=application%2Fjson&width=608&height=496&srs=EPSG%3A3857&bbox=1818436.9029043436%2C6143052.870708901%2C1824246.1170540168%2C6147791.966462584)

Who (and how) will do a validation of our current adaptation option calculation results procuded by Emikat? I see that @NICDDB provided an [example](https://github.com/clarity-h2020/csis/issues/154#issuecomment-661871632) of their calculations, so maybe we can just create a Study in CSIS covering the same area and use the same adaptation strategies as applied in his example? @NICDDB for which city (and location) did you run your example calculations?

**Comment by NICDDB** [comment]
Us example calculation was made in Naples on calata capodichino street, near the airport. 
I create a new study on CSIS where i've selected the same study area and i've add @patrickkaleta as owner.

**Comment by DenoBeno** [comment]
I am looking at this now:

https://csis-dev.myclimateservice.eu/study/25/step/5113/view/table

Sort by number of people living in the area and inspect the results,
please.

1) many densely populated areas have no AOs
2) changes in mortality are minimal.

I did choose the adaptation strategies that are mainly targeting floods,
but this doesn't look correct.

**Comment by mattia-leone** [comment]
Guys hope you can fix all this. At the moment I am working in finalizing D2.4 by Friday August 7th, since after that PLINIVS team will be on vacation and won't be available anymore until the end of the project, except answering the phone and reading emails from time to time.
I don't think that given the current status of impact and adaptation calculation I will be able to integrate the CSIS output in D2.4 as requested by @DenoBeno , but it's ok since we mainly focus on expert services there, and indeed we have calculated all impact and effect of adaptation using our local version of the CSIS heat and flood models.
Please let me know if you are willing to solve this in the next three days, in this case I can support you. 
I don't think we need validation, the method and calculations should work as @stefanon transferred them to @humerh , as proven by our DC1 results and already discussed in D3.3.
Wait for your considerations

**Comment by p-a-s-c-a-l** [comment]
You can also check AO calculation on [this map](https://csis-dev.myclimateservice.eu/apps/map-component/build/SynchronisedRiskAndImpactMap/?host=https://csis-dev.myclimateservice.eu&study_uuid=9359e741-df40-4bcd-9faf-2093b499c65c&study_area=POLYGON%20((16.346111%2048.223997,%2016.346111%2048.238634,%2016.376667%2048.238634,%2016.376667%2048.223997,%2016.346111%2048.223997))&emikat_id=3183&datapackage_uuid=2434ce93-93d4-4ca2-8618-a2de768d3f16&write_permissions=1&time_period=Baseline&emissions_scenario=Baseline&event_frequency=Rare&grouping_tag=taxonomy_term--hazards). Left is without AO, right side is with [this adaptation strategy](https://csis-dev.myclimateservice.eu/study/25/step/5112/view/table) applied.

![image](https://user-images.githubusercontent.com/1788226/89294231-803dea80-d65f-11ea-8a45-bcb9dae35e75.png)

**Comment by mattia-leone** [comment]
I guess that the most important thing is to have our toy working for the final review. About this, you can count on our support in September of course.
@p-a-s-c-a-l one question, can I make a new study in Napoli and test the adaptation strategies? 
I created a new study in the dev version (https://csis-dev.myclimateservice.eu/study/171/view/intro) and edited all the needed data but the calculation status is still on "not ready".. which is my mistake? 
Already tried in re-editing the area and changing datapackage. Added you and @DenoBeno as owner, please let me know what's wrong

**Comment by p-a-s-c-a-l** [comment]
@mattia-leone Sorry, EMIKAT calculation is disabled on **CSIS-DEV**, you have to https://csis.myclimateservice.eu. But I don't know if we have already latest version of CSIS-DEV published on CSIS @patrickkaleta ?

**Comment by mattia-leone** [comment]
ah ok, actually I supposed so, but saw Denis' study on Vienna with all the adaptation steps included, so I thought it was ready also for Napoli. Let me know if we can make it, as said I have time until Friday, then again in September to support final review preparation (which is much more important than deliverables when it comes to the CSIS!!)

**Comment by patrickkaleta** [comment]
> @mattia-leone Sorry, EMIKAT calculation is disabled on **CSIS-DEV**, you have to https://csis.myclimateservice.eu. But I don't know if we have already the latest version of CSIS-DEV published on CSIS @patrickkaleta ?

No, not yet. I think it will be available later today or tomorrow. Currently I'm testing the updated Group module after discovering [this issue](https://github.com/clarity-h2020/csis-technical-validation/issues/111#issuecomment-668016939) yesterday. The two Views in question have already been fixed, but since the Groups module plays such a major role in the CSIS I need to check the other Views as well, which might not all be covered by the CI automated testing.

**Comment by patrickkaleta** [comment]
> > @mattia-leone Sorry, EMIKAT calculation is disabled on **CSIS-DEV**, you have to https://csis.myclimateservice.eu. But I don't know if we have already the latest version of CSIS-DEV published on CSIS @patrickkaleta ?
> 
> No, not yet. I think it will be available later today or tomorrow. Currently I'm testing the updated Group module after discovering [this issue](https://github.com/clarity-h2020/csis-technical-validation/issues/111#issuecomment-668016939) yesterday. The two Views in question have already been fixed, but since the Groups module plays such a major role in the CSIS I need to check the other Views as well, which might not all be covered by the CI automated testing.

Synchronization from DEV to PROD is done, but I wasn't able to synchronize our GL-templates and the Study types (see #184 for details).

@mattia-leone nonetheless you should now be able to check the adaptation strategies in Napoli. I created [this Study](https://csis.myclimateservice.eu/study/195/view/intro) on PROD and made you the owner of it. Let me know if you have any troubles working with it.

**Comment by mattia-leone** [comment]
Hi Patrick, just read the message. I have now triggered the calculation and waiting for the results. 
Let you know my... feelings ;)

**Comment by mattia-leone** [comment]
<img width="1280" alt="Schermata 2020-08-05 alle 19 25 09" src="https://user-images.githubusercontent.com/56071027/89444028-667cd000-d751-11ea-8a64-e15adda8e09c.png">
A first strange thing is the flood local effect, which has some "holes" as in the screen attached. And the right map panel does not display any local results, but only the land use layers.

**Comment by mattia-leone** [comment]
apart from the holes, I think the flood local effect is not correct. It displays red cells up on the hill, while it should be the opposite. @stefanon maybe @patrickkaleta and/or @humerh have inverted the classification of some layers?

**Comment by mattia-leone** [comment]
<img width="1280" alt="Schermata 2020-08-05 alle 19 30 09" src="https://user-images.githubusercontent.com/56071027/89444520-181c0100-d752-11ea-9055-2bbddfd938e6.png">
also here right map panel does not display results. 
I see also some mismatches in the layers that are visualized here.
Risk/Impact maps should be limited to:
- Mortality rate increase (Heat Waves)
- Outdoor discomfort (Heat Waves) - I think you named the layer "Heat Waves impact" as from the impact calculation discomfort.doc.. my mistake I think it's more clear if we name it like this
- Economic impact on roads (Flood)
- Economic impact on residential buildings (Flood) 
- Economic impact on non-residential buildings (Flood) 

The layers "number of exposed persons" and "population density class" should be moved to the "Exposure" map tab.
The layers "heat damage probability" and "flood damage probability" I think can be removed, since they are not useful to end users

@patrickkaleta I hope you can easily fix this, tomorrow I will look at the Adaptation section

**Comment by p-a-s-c-a-l** [comment]
> And the right map panel does not display any local results, but only the land use layers.

The right **HC-LE** map is currently configured to show the **adapted scenario** which has not bee calculated. I'm, aware that for the user it's not obvious where and when adaption scenarios are are shown. The new [AO appraisal step](https://github.com/clarity-h2020/csis/issues/177) introduced some inconsistencies which I don't know how to resolve even at conceptual level. Anyway, I can change that if you prefer to see on the right map the non-adapted scenario **when no adapted scenario is available**, otherwise it shows the adapted one. 

> apart from the holes, I think the flood local effect is not correct. It displays red cells up on the hill, while it should be the opposite.

The map show only the parameter **PF_PROP**. There are many other HC-LE PF values available, which are not yet exposed via EMIKAT GeoServer: PF_STREAMS, PF_CL_STREAMS. PF_DELTA_ELEV, PF_CL_DELTA_ELEV, PF_RUN_OFF_AVERAGE, PF_CL_RUNOFF, PF_RAIN_MM: 80, PF_CL_RAIN. Perhaps it would help to show them on the map, too? For this @humerh would have to set-up the respective styles. 

>  I think it's more clear if we name it like this - Economic impact on roads (Flood) - Economic impact on residential buildings (Flood) - Economic impact on non-residential buildings (Flood) 

Hm, ATM there is only **one** 'flood impacts in euro' layer. 

 > Risk/Impact maps should be limited to: - Mortality rate increase (Heat Waves) - Outdoor discomfort (Heat Waves)

O.K. I will implement the changes in the respective [resource specification](https://csis-dev.myclimateservice.eu/node/1356) on CSIS-DEV. Please be aware that we have atm [some problems](https://github.com/clarity-h2020/csis/issues/184) with the synchronisation, so it might take till the end of the week for the changes to become available on CSIS-PROD.


> The layers "number of exposed persons" and "population density class" should be moved to the "Exposure" map tab.

To do this, the layers have to be made available in EMIKAT View [Exposure data (Population) calculated from EMIKAT](https://github.com/clarity-h2020/csis/wiki/Services-endpoints-(used-by-CSIS)#exposure-data-population-calculated-from-emikat) by @humerh. ATM it contains just POPULATION_TOTAL. Once this is done, I can update the respective [resource specification](https://csis-dev.myclimateservice.eu/node/754) on CSIS-DEV. Since I'm leaving the project on 12. August, I'll  also tell @therter how to make these changes in the resource metea-data.

> The layers "heat damage probability" and "flood damage probability" I think can be removed, since they are not useful to end users

O.K. flood damage probability is currently [not available](https://github.com/clarity-h2020/emikat/issues/49#issuecomment-669785758) anyway.

**Comment by mattia-leone** [comment]
hi @p-a-s-c-a-l thanks for the info. now i triggered the calculation, and indeed the right panel is showing the effect of adaptation strategies. 
I think it is ok, the most useful thing to display on the right panel, because it makes easy the comparisons. Maybe a text that explain that left scenario is no adaptation and the right is adaptation would increase user friendliness

this seems ok for heat waves but it should give some variation also on flood impact.. wonder why not

about the other points (sorry I don't know how to quote your sentences as you did with mine..):
- no, let's not add other HC-LE PF layers, it cause only confusion to the users
- in the table tab, I see that flood impacts in euro are separated (roads, res. buildings and non-res) so I guess they can be split also in the map, or not?
- the exposure map would be really much more useful if we can move there the layers "number of exposed persons" and "population density class" (which in any case should not stay in the risk/impact), hope this can be fixed

**Comment by p-a-s-c-a-l** [comment]
>  in the table tab, I see that flood impacts in euro are separated (roads, res. buildings and non-res) so I guess they can be split also in the map, or not?

If you are talking about PF_RESIDENTIAL_BUILDINGS;PF_NON_RESIDENTIAL_BUILDINGS;PF_ROADS_M2, I don't know if those are Economic impacts in €.  Perhaps @humerh can clarify. Those columns aren't available in EMIKAT GeoServer anyway, so @humerh would have to create the respective styles first.

**Comment by mattia-leone** [comment]
ok, I ask @humerh also to double check if the classification and style of the layer "flood damage class" is correct with respect to what indicated by @stefanon and me. I see almost everything red on that layer, while the damage in euro is much more diversified and more in line with the results I would expect in terms of impact.
I really don't understand the meaning of the "flood damage class" layer, as said in the risk/impact assessment for flood we should only have damage in euro (possibly split in the three subcategories of roads, residential and non-residential buildings).

**Comment by DenoBeno** [comment]
OK folks, we have a showstopper here: I have made new calculations in Linz and in Stockholm now and discomfort level is 5 everywhere and for every scenario. It's really silly, look at this:

![image](https://user-images.githubusercontent.com/2961790/89646996-17ec4480-d8bd-11ea-8f1b-4338fa99959a.png)

That's UTCI. And this is discomfort:

And the discomfort level in this table is NOT five everywhere! It's ranging from 2 to 4.

https://csis.myclimateservice.eu/study/179/step/4825/view/table

This table too. From 2 to 4.

https://csis.myclimateservice.eu/study/179/step/4828/view/table

My working assumption is that the discomfort is calculated properly but then somehow wrong values are taken for the maps. Same for the heat wave impact.

**Comment by DenoBeno** [comment]
One important question on impacts. What are these impacts that we calculate? Is it impact of a single event or cumulative impact of all events happening in 20 years (which is then nearer to risk).

When you think of it, the individual 20 year events should have much higher damage cost and mortality than 1y events. Like 10 times higher... And the sum of all 1 year events over 20 years could be both higher and lower than a single 20 y event...

**Comment by mattia-leone** [comment]
the impact calculation is on a single scenario representative of the selected reference event (frequent, rare, occasional) in the selected 30 year period. it is not the cumulative impact of all the event in the period

**Comment by p-a-s-c-a-l** [comment]
> My working assumption is that the discomfort is calculated properly but then somehow wrong values are taken for the maps.

The map uses the style HW_DISCOMFORT_LEVEL so it should show the same values as in the table. If yu clikc on a cell, You'll see that HW_DISCOMFORT_LEVEL  also ranges from 2 to 4. So it's probably just a problem of [styling](https://github.com/clarity-h2020/emikat/issues/54).

**Comment by DenoBeno** [comment]
Heinrich has fixed the maps now. At least the discomfort looks ok, didn't check the heat impacts part

**Comment by DenoBeno** [comment]
Now the table and the multi criteria steps are gone from this study.  That's really strange.

I looked at Linz map and here are two screenshots for past and for the worst case future scenario.  Both showing discomfort and heat impact,  both yearly events.

![Screenshot_20200808-125006](https://user-images.githubusercontent.com/2961790/89708287-02872100-d976-11ea-9899-2629d64e646d.png)
![Screenshot_20200808-124821](https://user-images.githubusercontent.com/2961790/89708289-061aa800-d976-11ea-8163-1c30ed87eb98.png)

It would seem that impact is wrong in the map.
